core/tests/ts-integration/src/modifiers/balance-checker.ts

```diff
@@ -194,44 +197,75 @@ class ShouldChangeBalance extends MatcherModifier {
  *
  * @param receipt Receipt of the transaction to extract fee from.
  * @param from Optional substitute to `receipt.from`.
+ * @param wallet Optional wallet if the api mode is modern
  * @returns Extracted fee
  */
-export function extractFee(receipt: zksync.types.TransactionReceipt, from?: string): Fee {
+export async function extractFee(
+    receipt: zksync.types.TransactionReceipt,
+    from?: string,
+    wallet?: zksync.Wallet
+): Promise<Fee> {
     from = from ?? receipt.from;
 
     const systemAccountAddress = '0x0000000000000000000000000000000000000000000000000000000000008001';
     // We need to pad address to represent 256-bit value.
     const fromAccountAddress = ethers.utils.hexZeroPad(ethers.utils.arrayify(from), 32);
-    // Fee log is one that sends money to the system contract account.
-    const feeLog = receipt.logs.find((log) => {
-        return log.topics.length == 3 && log.topics[1] == fromAccountAddress && log.topics[2] == systemAccountAddress;
-    });
-    if (!feeLog) {
-        throw {
-            message: `No fee log was found in the following transaction receipt`,
-            receipt
-        };
-    }
 
-    const feeAmount = ethers.BigNumber.from(feeLog.data);
+    const apiMode = process.env.EN_API_MODE ?? process.env.API_WEB3_JSON_RPC_API_MODE;
+
+    if (apiMode == 'modern') {
```

Comment:
Wouldn't this approach work for "legacy" API as well? If so, why do we need both?

Reply:
I am pretty sure that it will, but I think it is good to have separate tests for both API modes to understand and test difference between them

Reply:
But this isn't really a test, it's a modifier that is to be used by tests, so I'd expect it to do 1 thing in 1 way.
If you want to make sure that the second approach works, probably you can just extract it to the test which won't use this modifier and instead will make the calculations based on the 2nd approach.